### PR TITLE
Fixes refresh related bugs

### LINKED ISSUE #2361: UI window location creates an invalid url
When you execute a search, the window location changes from zipkin/ to zipkin (without slash), which results in a 404 if you execute that.

### LINKED ISSUE #2443: zipkin ui refresh request again response no anything
docker-zipkin version:2.12.5

docker-zipkin version : 2.11.12 is ok. other version not test

deploy env: kubernetes v1.13.1

simple deploy yaml:
```yaml
env:
            - name: "STORAGE_TYPE"
              value: "elasticsearch"
            - name: "ES_HOSTS"
              value: "http://192.168.66.62:9200"
            - name: "ES_MAX_REQUESTS"
              value: "64"
            - name: "ES_INDEX"
              value: "rdc_cloud_zipkin_trace"
            - name: "ElasticsearchStorage.Builder.index"
              value: "rdc_cloud_zipkin_trace"
            - name: "ES_DATE_SEPARATOR"
              value: "."
            - name: "ElasticsearchStorage.Builder.dateSeparator"
              value: "."
            - name: "ES_INDEX_SHARDS"
              value: "5"
            - name: "ES_INDEX_REPLICAS"
              value: "1"
            - name: "KAFKA_BOOTSTRAP_SERVERS"
              value: "192.168.67.21:9092,192.168.67.22:9092,192.168.67.23:9092"
            - name: "KAFKA_TOPIC"
              value: "rdc_cloud_zipkin_trace"
            - name: "KAFKA_GROUP_ID"
              value: "rdc_cloud_zipkin_trace_zipkin_group"
            - name: "QUERY_PORT"
              value: "9411"
            - name: "ZIPKIN_UI_DEPENDENCY_LOW_ERROR_RATE"
              value: "0.3"
            - name: "ZIPKIN_UI_DEPENDENCY_HIGH_ERROR_RATE"
              value: "0.75"
            - name: "ZIPKIN_UI_LOGS_URL"
              value: "http://devcloud.i139.cn/kibana/app/kibana#/discover?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(_source),index:'b7e371c0-4638-11e9-bad2-61d86dd48551',interval:auto,query:(language:lucene,query:'{traceId}'),sort:!(timestamp_millis,desc))"
            - name: "JAVA_OPTS"
              value: "-Xms1G -Xmx1G -Dlogging.level.zipkin=WARN -Dlogging.level.zipkin2=WARN"
```

First request zipkin ui every thing is OK.

1111552621436_ pic_hd

But when refresh zipkin ui in chrome browser, chrome browser print no thing,however the response is 200, And use curl http://zipkin-ui-url:9411 response normal data forever:

1121552621504_ pic_hd

1101552621381_ pic_hd

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This fixes two refresh related bugs:
* cache-control was returning 200 not 304 due to a bug
* /zipkin wasn't redirecting to /zipkin/ (only affects lens)

Fixes #2443
Fixes #2361

### CODE REVIEW

File: zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java [hunk 1]
```diff
@@ -77,15 +77,6 @@
 @EnableConfigurationProperties(ZipkinUiProperties.class)
 @ConditionalOnProperty(name = "zipkin.ui.enabled", matchIfMissing = true)
 class ZipkinUiAutoConfiguration {
-  static final HttpHeaders CACHE_YEAR =
-    HttpHeaders.of(CACHE_CONTROL, "max-age=" + TimeUnit.DAYS.toSeconds(365));
-  static final HttpHeaders CONFIG_HEADERS = HttpHeaders.of(HttpStatus.OK)
-    .add(HttpHeaders.of(CONTENT_TYPE, "application/json"))
-    .add(HttpHeaders.of(CACHE_CONTROL, "max-age=" + TimeUnit.MINUTES.toSeconds(10)));
-  static final HttpHeaders INDEX_HEADERS = HttpHeaders.of(HttpStatus.OK)
```
Comment (codefromthecrypt) [inline zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java:85]: bug was this `HttpHeaders.of(HttpStatus.OK)` which was a hack no longer needed as cacheControl can be done directly now.

**Review by anuraaga — APPROVED** [review]
Thanks for fixing, is it possible to add a test for the codepath?

**Comment by codefromthecrypt** [comment]
wrt testing yep. I can do it on the plane tomorrow as it will take more than a few minutes

File: zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java [hunk 2]
```diff
@@ -130,27 +121,30 @@ String processedIndexHtml(Resource indexHtml) {
...
+    IndexSwitchingService indexSwitchingService) throws IOException {
+    ServerCacheControl maxAgeYear =
+      new ServerCacheControlBuilder().maxAgeSeconds(TimeUnit.DAYS.toSeconds(365)).build();
     Service<HttpRequest, HttpResponse> uiFileService =
-      new AddHttpHeadersService(HttpFileService.forClassPath("zipkin-ui")
-        .orElse(HttpFileService.forClassPath("zipkin-lens")), CACHE_YEAR);
-
-
+      HttpFileServiceBuilder.forClassPath("zipkin-ui").cacheControl(maxAgeYear).build()
+        .orElse(HttpFileServiceBuilder.forClassPath("zipkin-lens").cacheControl(maxAgeYear).build());
```
Comment (devinsba) [inline zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java:141]: Our JS/CSS bundles get a hash appended right?

**Review by devinsba — APPROVED** [review]

**Review by shakuzen — APPROVED** [review]

File: zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java (hunk 2 above)
Comment (codefromthecrypt) [reply zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java:141]: just added a comment that should have been there :)

**Comment by codefromthecrypt** [comment]
will put in the integration tests during my flight. see ya

**Comment by codefromthecrypt** [comment]
2.12.6 on the way

**Pull request merged**